code/modules/mob/living/basic/constructs/artificer.dm

```diff
@@ -33,70 +29,33 @@
 
 	can_repair = TRUE
 	can_repair_self = TRUE
+	smashes_walls = TRUE
 	///The health HUD applied to this mob.
 	var/health_hud = DATA_HUD_MEDICAL_ADVANCED
 
-/mob/living/simple_animal/hostile/construct/artificer/Initialize(mapload)
+/mob/living/basic/construct/artificer/Initialize(mapload)
 	. = ..()
+	AddElement(/datum/element/ai_retaliate)
 	var/datum/atom_hud/datahud = GLOB.huds[health_hud]
 	datahud.show_to(src)
 
-/mob/living/simple_animal/hostile/construct/artificer/Found(atom/thing) //what have we found here?
-	if(!isconstruct(thing)) //is it a construct?
-		return FALSE
-	var/mob/living/simple_animal/hostile/construct/cultie = thing
-	if(cultie.health < cultie.maxHealth) //is it hurt? let's go heal it if it is
-		return TRUE
-
-/mob/living/simple_animal/hostile/construct/artificer/CanAttack(atom/the_target)
-	if(see_invisible < the_target.invisibility)//Target's invisible to us, forget it
-		return FALSE
-	if(Found(the_target) || ..()) //If we Found it or Can_Attack it normally, we Can_Attack it as long as it wasn't invisible
-		return TRUE //as a note this shouldn't be added to base hostile mobs because it'll mess up retaliate hostile mobs
-	return FALSE
-
-/mob/living/simple_animal/hostile/construct/artificer/MoveToTarget(list/possible_targets)
-	..()
-	if(!isliving(target))
-		return
-
-	var/mob/living/victim = target
-	if(isconstruct(victim) && victim.health >= victim.maxHealth) //is this target an unhurt construct? stop trying to heal it
-		LoseTarget()
-		return
-	if(victim.health <= melee_damage_lower+melee_damage_upper) //ey bucko you're hurt as fuck let's go hit you
-		retreat_distance = null
-		minimum_distance = 1
-
-/mob/living/simple_animal/hostile/construct/artificer/Aggro()
-	..()
-	if(isconstruct(target)) //oh the target is a construct no need to flee
-		retreat_distance = null
-		minimum_distance = 1
-
-/mob/living/simple_animal/hostile/construct/artificer/LoseAggro()
-	..()
-	retreat_distance = initial(retreat_distance)
-	minimum_distance = initial(minimum_distance)
-
-/mob/living/simple_animal/hostile/construct/artificer/hostile //actually hostile, will move around, hit things, heal other constructs
-	AIStatus = AI_ON
-	environment_smash = ENVIRONMENT_SMASH_STRUCTURES //only token destruction, don't smash the cult wall NO STOP
+/// Hostile NPC version. Heals nearby constructs and cult structures, avoids targets that aren't extremely hurt.
+/mob/living/basic/construct/artificer/hostile
+	ai_controller = /datum/ai_controller/basic_controller/artificer
+	smashes_walls = FALSE
 
 /////////////////////////////Artificer-alts/////////////////////////
```

Comment:
Minor but can we remove the big comment headers to just use `//` ?
It fucks with autodocs